### PR TITLE
Remove SimpleLogProcessor from log benchmark

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

Nit change to remove SimpleLogProcessor from the log benchmark. The benchmark is mainly for Logger and LogRecord, the result shouldn't be affected by the [mutex lock ](https://github.com/open-telemetry/opentelemetry-rust/blob/383c7d13b6fbda22dff993a39ce4f3aa1ff0da62/opentelemetry-sdk/src/logs/log_processor.rs#L79) for SimpleLogProcessor. Simple*Processor is only for testing purposes and can be avoided in general in bench tests. Or else, there can be a separate test for it if required.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/benches/log.rs [hunk 1]
```diff
@@ -16,26 +16,62 @@ use opentelemetry::trace::Tracer;
...
     }
 }
 
+#[derive(Debug)]
+struct NoopProcessor {
+    _exporter: Box<dyn LogExporter>,
+}
+
+impl NoopProcessor {
+    fn new(exporter: Box<dyn LogExporter>) -> Self {
```
Comment (cijothomas) [inline opentelemetry-sdk/benches/log.rs:40]: why does this need an exporter, if it is noop anyway?

**Review by cijothomas — APPROVED** [review]
LGTM (Please cleanup the NoOp processor to be NoOp, without even requiring an exporter)

File: opentelemetry-sdk/benches/log.rs (hunk 1 above)
Comment (lalitb) [reply opentelemetry-sdk/benches/log.rs]: Good point. Removed.

**Comment by lalitb** [comment]
CI errors due to updated rust toolchain. Will create a separate PR for this.
```
    Checking opentelemetry-zipkin v0.22.0 (/home/runner/work/opentelemetry-rust/opentelemetry-rust/opentelemetry-zipkin)
error: doc list item missing indentation
 --> opentelemetry-zipkin/src/propagator/mod.rs:6:5
  |
6 | //!    b3: {trace_id}-{span_id}-{sampling_state}-{parent_span_id}
  |     ^^^
  |
  = help: if this is supposed to be its own paragraph, add a blank line
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_lazy_continuation
  = note: `-D clippy::doc-lazy-continuation` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(clippy::doc_lazy_continuation)]`
help: indent this line
  |
6 | //!     b3: {trace_id}-{span_id}-{sampling_state}-{parent_span_id}
  |        +

error: doc list item missing indentation
 --> opentelemetry-zipkin/src/propagator/mod.rs:8:5
  |
8 | //!    X-B3-TraceId: {trace_id}
  |     ^^^
  |
  = help: if this is supposed to be its own paragraph, add a blank line
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_lazy_continuation
help: indent this line
  |
8 | //!     X-B3-TraceId: {trace_id}
  |        +

error: doc list item missing indentation
 --> opentelemetry-zipkin/src/propagator/mod.rs:9:5
  |
9 | //!    X-B3-ParentSpanId: {parent_span_id}
  |     ^^^
  |
  = help: if this is supposed to be its own paragraph, add a blank line
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_lazy_continuation
help: indent this line
  |
9 | //!     X-B3-ParentSpanId: {parent_span_id}
  |        +

error: doc list item missing indentation
  --> opentelemetry-zipkin/src/propagator/mod.rs:10:5
   |
10 | //!    X-B3-SpanId: {span_id}
   |     ^^^
   |
   = help: if this is supposed to be its own paragraph, add a blank line
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_lazy_continuation
help: indent this line
   |
10 | //!     X-B3-SpanId: {span_id}
   |        +

error: doc list item missing indentation
  --> opentelemetry-zipkin/src/propagator/mod.rs:11:5
   |
11 | //!    X-B3-Sampled: {sampling_state}
   |     ^^^
   |
   = help: if this is supposed to be its own paragraph, add a blank line
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_lazy_continuation
help: indent this line
   |
11 | //!     X-B3-Sampled: {sampling_state}
   |        +

error: doc list item missing indentation
  --> opentelemetry-zipkin/src/propagator/mod.rs:12:5
   |
12 | //!    X-B3-Flags: {debug_flag}
   |     ^^^
   |
   = help: if this is supposed to be its own paragraph, add a blank line
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_lazy_continuation
help: indent this line
   |
12 | //!     X-B3-Flags: {debug_flag}
   |        +

error: could not compile `opentelemetry-zipkin` (lib) due to 6 previous errors
warning: build failed, waiting for other jobs to finish...
error: doc list item missing indentation
   --> opentelemetry-zipkin/src/propagator/mod.rs:6:5
    |
6   | //!    b3: {trace_id}-{span_id}-{sampling_state}-{parent_span_id}
    |     ^^^
    |
    = help: if this is supposed to be its own paragraph, add a blank line
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_lazy_continuation
note: the lint level is defined here
   --> opentelemetry-zipkin/src/lib.rs:203:24
    |
203 | #![cfg_attr(test, deny(warnings))]
    |                        ^^^^^^^^
    = note: `#[deny(clippy::doc_lazy_continuation)]` implied by `#[deny(warnings)]`
help: indent this line
    |
6   | //!     b3: {trace_id}-{span_id}-{sampling_state}-{parent_span_id}
    |        +

error: could not compile `opentelemetry-zipkin` (lib test) due to 6 previous errors
```

**Pull request merged**